src/main/java/uk/ac/ebi/embl/converter/gff3toff/GFF3Mapper.java

```diff
@@ -38,71 +34,95 @@ public class GFF3Mapper {
                     String accessionId = accession.substring(0, accession.lastIndexOf('.'));
                     entry.setPrimaryAccession(accessionId);
                     Location location = this.locationFactory.createLocalRange(reg.start(), reg.end());
-                    Order<Location> compoundJoin = new Order();
+                    Join<Location> compoundJoin = new Join<>();
                     compoundJoin.addLocation(location);
                     sourceFeature.setLocations(compoundJoin);
                 }
             }
             entry.addFeature(sourceFeature);
         }
 
-        for (GFF3Feature feature : annotation.getFeatures()) {
-            if (feature.getId().isPresent()) {
-                parentFeatures.put(feature.getId().get(), feature);
+        for (GFF3Feature gff3Feature : gff3Annotation.getFeatures()) {
+            if (gff3Feature.getId().isPresent()) {
+                parentFeatures.put(gff3Feature.getId().get(), gff3Feature);
             }
 
-            entry.addFeature(mapGFF3Feature(feature));
+            mapGFF3Feature(gff3Feature);
 
-
-            for (GFF3Feature childFeature : feature.getChildren()) {
-                Feature ffChildFeature = mapGFF3Feature(childFeature);
-                entry.addFeature(ffChildFeature);
+            for (GFF3Feature childFeature : gff3Feature.getChildren()) {
+                mapGFF3Feature(childFeature);
             }
         }
 
         return entry;
     }
 
-    private Feature mapGFF3Feature(GFF3Feature gff3Feature) {
+    private void mapGFF3Feature(GFF3Feature gff3Feature) {
 
         Map<String, String> attributes = gff3Feature.getAttributes();
         Collection<Qualifier> qualifiers = mapGFF3Attributes(attributes);
-
-        CompoundLocation<Location> locations = mapGFF3Location(gff3Feature);
-
+        String featureHashId = attributes.getOrDefault("ID", String.valueOf(gff3Feature.hashCode()));
         String featureType = gff3Feature.getName();
-        Feature ffFeature = this.featureFactory.createFeature(featureType);
-        ffFeature.setLocations(locations);
-        ffFeature.addQualifiers(qualifiers);
+
+        Location location = mapGFF3Location(gff3Feature);
+        Feature ffFeature = ffFeatures.get(featureHashId);
+        if (ffFeature != null) {
+            CompoundLocation<Location> locations = ffFeature.getLocations();
+            // If the compoundlocation isComplement but the new location we are adding is not complement
+            // we need to restructure the locations that it contains
+            // QUESTION: Is this ever happen? AFAIK the syntax would allow this, but it is nonsensical.
+            if (locations.isComplement() && !location.isComplement()) {
+                locations.getLocations().forEach((l) -> location.setComplement(true));
+                locations.setComplement(false);
+            } else if (location.isComplement() && location.isComplement()) {
```

Comment:
Below code is not  correct:
`} else if (location.isComplement() && location.isComplement()) {`
Must be updated to:
`} else if (parentFeatureLocation.isComplement() && location.isComplement()) {`